File: server/controllers/adminController.ts
import { Request, Response } from 'express';
import pool from '../config/db';

export const getUsers = async (req: Request, res: Response) => {
  try {
    const result = await pool.query('SELECT id, email, role, name, created_at FROM users');
    res.json(result.rows);
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
};

export const getSystemStats = async (req: Request, res: Response) => {
  try {
    const users = await pool.query('SELECT count(*) FROM users');
    const patients = await pool.query('SELECT count(*) FROM patients');
    const admissions = await pool.query('SELECT count(*) FROM admissions WHERE status = \'admitted\'');
    res.json({
      totalUsers: parseInt(users.rows[0].count),
      totalPatients: parseInt(patients.rows[0].count),
      activeAdmissions: parseInt(admissions.rows[0].count)
    });
  } catch (err: any) {
    res.status(500).json({ error: err.message });
  }
};

File: server-java/src/main/java/com/mediflow/api/controller/AdminController.java
package com.mediflow.api.controller;

import com.mediflow.api.model.User;
import com.mediflow.api.repository.*;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.*;

@RestController
@RequestMapping("/api/admin")
@PreAuthorize("hasRole('Admin')")
public class AdminController {

    @Autowired
    private UserRepository userRepository;

    @Autowired
    private PatientRepository patientRepository;

    @Autowired
    private AdmissionRepository admissionRepository;

    @GetMapping("/users")
    public ResponseEntity<?> getUsers() {
        try {
            List<User> list = userRepository.findAll();
            
            // Map to Express-compatible JSON (exclude password hash for safety)
            List<Map<String, Object>> response = new ArrayList<>();
            for (User u : list) {
                Map<String, Object> item = new HashMap<>();
                item.put("id", u.getId());
                item.put("email", u.getEmail());
                item.put("role", u.getRole());
                item.put("name", u.getName());
                item.put("created_at", u.getCreatedAt());
                
                response.add(item);
            }
            
            return ResponseEntity.ok(response);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(Map.of("error", e.getMessage()));
        }
    }

    @GetMapping("/stats")
    public ResponseEntity<?> getSystemStats() {
        try {
            long totalUsers = userRepository.count();
            long totalPatients = patientRepository.count();
            // Count where status = 'Admitted'
            long activeAdmissions = admissionRepository.countByStatusIgnoreCase("Admitted");

            Map<String, Object> response = new HashMap<>();
            response.put("totalUsers", totalUsers);
            response.put("totalPatients", totalPatients);
            response.put("activeAdmissions", activeAdmissions);

            return ResponseEntity.ok(response);
        } catch (Exception e) {
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(Map.of("error", e.getMessage()));
        }
    }
}
